awd/dynamodb.py

```diff
@@ -71,23 +82,40 @@ def retrieve_doi_items_from_database(self, doi_table):
     def retry_attempts_exceeded(self, doi_table, doi, retry_threshold):
         """Validate whether a DOI has exceeded the retry threshold."""
         validation_status = False
```

Comment:
Going back to this: it will be useful to rename to something more concrete such as attempts_exceeded.